Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Dockerfile for automated build of signal-cli
+#
+# Refer to the signal-cli github pages for detailed Usage.
+#
+FROM gradle:3.5-jdk7-alpine
+
+LABEL maintainer "Kayvan Sylvan <kayvansylvan@gmail.com>"
+
+USER root
```

Comment:
That's exactly what happens. Only the build happens as root, to install the needed components in the image. Further down in the Dockerfile we change the user to `gradle` before running `signal-cli`

When you run a container, it runs as the `gradle` user in the container but you can always change that on the command line.